src/Homie/Config.cpp

```diff
@@ -85,11 +85,11 @@ bool Config::load() {
   const char* reqWifiDns2 = reqWifi["dns2"] | "";
 
   uint16_t reqMqttPort = reqMqtt["port"] | DEFAULT_MQTT_PORT;
-  bool reqMqttSsl = reqMqtt["ssl"] | false;
+//  bool reqMqttSsl = reqMqtt["ssl"] | false;
```

Comment:
This would break MQTT over SSL for me.
As this PR contains a mixture of multiple things:
- ota_updater changes
- LittleFS support
- broken SSL support
- changes on WiFi connect behaviour
we should split this up into multiple smaller PRs covering each a single topic. I will take care about a compiler flags driven switch between SPIFFS and LittleFS.

Reply:
Thank you, for the review. Maybe the ota_updater changes are caused by merging my changes back into his fork?
I also started working on the LittleFS change but stopped when it came to the async web server, since it doesn't support LittleFS. IMO LittleFS can right now only be supported with HOMIE_CONFIG=0. When the web server is used, we should default to SPIFFS.

Reply:
I did not have a look at the commits and commit history but I think, we should try to keep Git history a little bit clean. For me code changes are already hard to follow.
Maybe I can make compile fail if LittleFS is selected without disabling the web server.

Reply:
It's better with a clean start. If you give it a go using this PR as inspiration, we can close this one in a few days.
Failing the compilation should be easy with a set of nested #if(n)defs